application/classes/Ushahidi/Repository/Form/Stats.php

```diff
@@ -248,5 +310,98 @@ private function targetedSurveyStateJoin($query) {
 			->on('contacts.id', '=', 'targeted_survey_state.contact_id')
 			->join('posts', 'INNER')
 			->on('posts.id', '=', 'targeted_survey_state.post_id');
-	}
+    }
+    
+    public function getSurveyType($form_id)
+    {
+        $query = DB::select('targeted_survey')
+        ->from('forms')
+        ->where('id', '=', $form_id);
+        $results = $query->execute($this->db);
+        return $results->as_array();
+    }
+
+    public function getPostCountByDataSource($form_id, $created_after, $created_before)
+    {
+        if ($created_after) {
+            $created_after = strtotime($created_after);
+        }
+        if ($created_before) {
+            $created_before = strtotime($created_before);
+        }
+        $dataSourceCounts = $this->queryByDataSource($form_id, $created_after, $created_before);
+        $result = [
+            'sms' => $dataSourceCounts['sms'],
+            'email' => $dataSourceCounts['email'],
+            'twitter' => $dataSourceCounts['twitter'],
+            'web' => $this->queryForWeb($form_id, $created_after, $created_before),
+            'all' => $this->queryForAllPosts($form_id, $created_after, $created_before)
+        ];
+        return $result;
+    }
+
+    private function queryByDataSource($form_id, $created_after, $created_before)
+    {
+        $query = DB::select('messages.type', [DB::expr('COUNT(messages.id)'), 'total'])
+            ->from('posts')
+            ->join('messages', 'INNER')
+            ->on('messages.post_id', '=', 'posts.id')
+            ->where('posts.form_id', '=', $form_id)
+            ->group_by('messages.type');
+        
+        if ($created_after) {
+            $query->where('messages.created', '>=', $created_after);
+        }
+        if ($created_before) {
+            $query->where('messages.created', '>=', $created_after);
+        }
+        $result = $query
+            ->execute($this->db);
+        $ret = ['sms' => 0, 'email' => 0, 'twitter' => 0];
+        foreach ($result->as_array() as $item) {
+            if ($item['type'] === 'sms') {
+                $ret['sms'] = $item['total'];
+            } elseif ($item['type'] === 'email') {
+                $ret['email'] = $item['total'];
+            } elseif ($item['type'] === 'twitter') {
+                $ret['twitter'] = $item['total'];
+            }
+        }
+        return $ret;
+    }
+
+    private function queryForWeb($form_id, $created_after, $created_before)
+    {
+        $query = DB::select([DB::expr('COUNT(posts.id)'), 'total'])
+            ->from('posts')
+            ->join('messages', 'LEFT')
+            ->on('messages.post_id', '=', 'posts.id')
+            ->where('messages.post_id', 'is', null)
+            ->where('posts.form_id', '=', $form_id);
+        if ($created_after) {
+            $query->where('posts.created', '>=', $created_after);
+        }
+        if ($created_before) {
```

Comment:
original code has a bug. Should be <= created_before https://github.com/ushahidi/platform/pull/3219/files see this PR